logger/logger.go

```diff
@@ -0,0 +1,131 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logger
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+const (
+	// Using RFC3339Nano as the timestamp format for logs as prometheus scrape interval can be in milliseconds.
+	timestampFormat = time.RFC3339Nano
+	// Prefix of the name of the log file to store task ranker logs.
+	// This will be suffixed with the timestamp, associated with creating the file, to obtain the log filename.
+	taskRankerLogFilePrefix = "task_ranker_logs"
+	// Prefix of the name of the log file to store task ranking results.
+	// This will be suffixed with the timestamp, associated with creating the file, to obtain the log filename.
+	taskRankingResultsLogFilePrefix = "task_ranking_results"
+	// Giving everyone read and write permissions to the log files.
+	logFilePermissions = 0666
+)
+
+// instantiating a Logger to be used. This instance is configured and maintained locally.
+var log = logrus.New()
+
+// createTaskRankerLogFile creates the log file to which task ranker logs are persisted.
+func createTaskRankerLogFile(now time.Time) (*os.File, error) {
+	filename := fmt.Sprintf("%s_%v.log", taskRankerLogFilePrefix, now.UnixNano())
+	taskRankerLogFile, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY, logFilePermissions)
+	if err != nil {
+		err = errors.Wrap(err, "failed to create task ranker operations log file")
+	}
+	return taskRankerLogFile, err
+}
+
+// createTaskRankingResultsLogFile creates the log file to which task ranking results are persisted.
+func createTaskRankingResultsLogFile(now time.Time) (*os.File, error) {
+	filename := fmt.Sprintf("%s_%v.log", taskRankingResultsLogFilePrefix, now.UnixNano())
+	taskRankingResultsLogFile, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY, logFilePermissions)
+	if err != nil {
+		err = errors.Wrap(err, "failed to create task ranker log file")
+	}
+	return taskRankingResultsLogFile, err
+}
+
+var taskRankerLogFile *os.File
+var taskRankingResultsLogFile *os.File
+
+// Configure the logger. To be prevented task ranker logs from mixing with the logs of the application
+// that is using it, logging to the console is disabled and instead hooks that redirect logs to corresponding
+// log files are attached to the logger.
+func Configure() error {
+	// Disabling log to stdout.
+	log.SetOutput(ioutil.Discard)
+	// Setting highest log level.
+	log.SetLevel(logrus.InfoLevel)
+
+	// Creating the log files.
+	now := time.Now()
+	var err error
+
+	taskRankerLogFile, err = createTaskRankerLogFile(now)
```

Comment:
taskRankerLogFile pointer already set within the function body. Do we need the assignment again?

Reply:
you are right. Currently, however there is one package scoped variable and another local to the function. I'm going to get rid of the local version and just initialize the package scoped variable.